parlai/agents/starspace/starspace.py

```diff
@@ -31,14 +31,14 @@ class StarspaceAgent(Agent):
     """
 
     OPTIM_OPTS = {
-        'adadelta': optim.Adadelta,
-        'adagrad': optim.Adagrad,
+        'adadelta': optim.Adadelta,  # type: ignore
```

Comment:
should we just have this be in a utils file somewhere? looks like seq2seq uses it as does starspace

Reply:
I'm not sure if I care to refactor. I'm going to retire that seq2seq soon enough, and torch agent is the canonical way to handle this.